android-sdk-simplexml/src/main/java/com/microsoft/hsg/android/simplexml/HealthVaultFileSettings.java

```diff
@@ -22,198 +22,221 @@
-        return properties.getProperty("service-url");
-    }
-    
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#setServiceUrl(java.lang.String)
-     */
-    public void setServiceUrl(String url) {
-        properties.setProperty("service-url", url);
-    }
-
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#getShellUrl()
-     */
-    public String getShellUrl() {
-        return properties.getProperty("shell-url");
-    }
-
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#setShellUrl(java.lang.String)
-     */
-    public void setShellUrl(String url) {
-        properties.setProperty("shell-url", url);
-    }
-
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#setAppId(java.lang.String)
-     */
-    public void setAppId(String appId) {
-        properties.setProperty("appid", appId);
-    }
-    
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#getAppId()
-     */
-    public String getAppId() {
-        return properties.getProperty("appid");
-    }
-
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#setAuthenticationSecret(java.lang.String)
-     */
-    public void setAuthenticationSecret(String secret) {
-        properties.setProperty("authentication-secret", secret);
-
-    }
-    
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#getConnectionStatus()
-     */
-    public HealthVaultApp.ConnectionStatus getConnectionStatus()
-    {
-        if (properties.containsKey("connected")) {
-            return HealthVaultApp.ConnectionStatus.Connected;
-        }
-        else {
-            return HealthVaultApp.ConnectionStatus.NotConnected;
-        }
-    }
-    
-    /* (non-Javadoc)
-     * @see com.microsoft.hsg.android.HealthVaultSettings#setConnectionStatus(com.microsoft.hsg.android.HealthVaultService.ConnectionStatus)
-     */
-    public void setConnectionStatus(HealthVaultApp.ConnectionStatus status) {
-        if (status == HealthVaultApp.ConnectionStatus.Connected) {
-            properties.setProperty("connected", "true");
-        } else {
-        	properties.remove("connected");
-        }
-    }
-    
-    public boolean getIsMRA() {
-		return Boolean.valueOf(properties.getProperty("is-mra"));
+		mProperties.setProperty(Constants.IsMultiInstanceAware, String.valueOf(isAware));
+	}
+	/* (non-Javadoc)
+	 * @see com.microsoft.hsg.android.HealthVaultSettings#getAuthenticationSecret()
+	 */
+	public String getAuthenticationSecret() {
+		return mProperties.getProperty(Constants.AuthenticationSecret);
+	}
+
+	/* (non-Javadoc)
+	 * @see com.microsoft.hsg.android.HealthVaultSettings#setMasterAppId(java.lang.String)
+	 */
+	public void setMasterAppId(String appId) {
+		mProperties.setProperty(Constants.MasterAppId, appId);
+	}
+
+	public void setSessionExpiration() {
+		mExperation = DateTime.now().plusHours(4);
```

Comment:
This is for the session token expiration defined by the server

---
In reply to: [118799423](https://github.com/Microsoft/healthvault-java-sdk/pull/5#discussion_r118799423) [](ancestors = 118799423)